### PR TITLE
PoC: Write only remote password

### PR DESCRIPTION
**Pull request by daviddavis**
[noissue]

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/tests/functional/api/using_plugin/test_crud_repos.py [hunk 1]
```diff
@@ -251,6 +256,31 @@ def test_update(self):
...
+        monitor_task(response.task)
+        exc = cli_client.run(["pulpcore-manager", "shell", "-c", shell_cmd])
+        self.assertEqual(exc.stdout.rstrip("\n"), "new")
+
+        # test a PUT request with a new password
+        remote_update = FileFileRemote(name="test_pass", url="http://", password="changed")
+        response = self.remotes_api.update(href, remote_update)
+        monitor_task(response.task)
+        exc = cli_client.run(["pulpcore-manager", "shell", "-c", shell_cmd])
+        self.assertEqual(exc.stdout.rstrip("\n"), "changed")
```
Comment (fao89) [inline pulpcore/tests/functional/api/using_plugin/test_crud_repos.py:282]: sweet!

File: pulpcore/tests/functional/api/using_plugin/test_crud_repos.py [hunk 2]
```diff
@@ -251,6 +256,31 @@ def test_update(self):
...
+        remote = self.remotes_api.create({"name": "test_pass", "url": "http://", "password": "new"})
+        href = remote.pulp_href
+        uuid = re.search(r"/pulp/api/v3/remotes/file/file/([\w-]+)/", href).group(1)
+        shell_cmd = (
+            f"import pulpcore; print(pulpcore.app.models.Remote.objects.get(pk='{uuid}').password)"
+        )
+
+        self.addCleanup(self.remotes_api.delete, href)
+
+        # test a PUT request without a password
```
Comment (bmbouter) [inline pulpcore/tests/functional/api/using_plugin/test_crud_repos.py:270]: Can each of these groupings become their own tests? Does that make sense to do?

**Comment by bmbouter** [comment]
I believe all the possibly sensitive info is [on the Remote MasterModel](https://github.com/pulp/pulpcore/blob/a70ab191d0e431469c04591b6dfaf177a7e6001f/pulpcore/app/models/repository.py#L183-L220), is that right?

### Potentially needing to be write_only
* username
* password
* proxy_url
* proxy_username - not yet merged, but soon https://pulp.plan.io/issues/8167
* proxy_password - not yet merged, but soon https://pulp.plan.io/issues/8167
* client_cert
* client_key

### bmbouter's recommendations
* username - yes because this can be significant
* password - yes, definitely
* proxy_url - no, but it's debatable. Users could set username/password in this. Perhaps we could have the validation disable users from doing that. That would be good. Being able to read back the proxy_url is important for users I think.
* proxy_username - yes because this can be significant
* proxy_password - yes, definitly
* client_cert - no, certs aren't useful without the key
* client_key - yes, definitly

What do you think?

What do you think?

**Comment by daviddavis** [comment]
Makes sense. I filed an issue and added it to the pulpcore agenda topic.

https://pulp.plan.io/issues/8202

**Pull request closed without merging**